ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OptimizeQueueService.java

```diff
@@ -561,19 +561,22 @@ public OptimizeTask poll(JobId jobId, final String attemptId, long waitTime) {
           }
         } else {
           if (tables.contains(task.getTableIdentifier())) {
+            TableTaskHistory tableTaskHistory;
             try {
               // load files from sysdb
               task.setFiles();
+              // update max execute time
+              task.setMaxExecuteTime();
+              tableTaskHistory = task.onExecuting(jobId, attemptId);
             } catch (Exception e) {
               task.clearFiles();
               LOG.error("{} failed to load files from sysdb, try put task back into queue", task.getTaskId(), e);
               if (!tasks.offer(task)) {
+                LOG.error("{} failed to put task back into queue", task.getTaskId());
                 task.onFailed(new ErrorMessage(System.currentTimeMillis(), "failed to put task back into queue"), 0);
               }
+              continue;
```

Comment:
We should sleep for like 1 second, before `continue` (poll from the tasks again).

Reply:
OK